### PR TITLE
fontforge: fix @rpath on arm64

### PR DESCRIPTION
**Pull request by radex**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/fontforge.rb [hunk 1]
```diff
@@ -43,6 +43,14 @@ def install
 
       # The "extras" built above don't get installed by default.
       bin.install Dir["bin/*"].select { |f| File.executable? f }
+
+      on_macos do
+        # Fix @rpath issue on arm64
+        system "install_name_tool", "-change",
```
Comment (carlocab) [inline Formula/fontforge.rb:49]: Does using `CMAKE_INSTALL_RPATH` or `CMAKE_INSTALL_NAME_DIR` not work? If not, then this is probably a bug that needs to be reported upstream.

**Comment by radex** [comment]
I've ran into trouble with `brew audit`.

I used `install_name_tool` to fix up dylib reference and it works. I tried these two alternatives that I think should do the exact same thing, but they fail:

```rb
        MachO::Tools.change_install_name("#{bin}/fontforge",
                                          "@rpath/libfontforge.4.dylib",
                                          "#{lib}/libfontforge.4.dylib")
```

```rb
        macho = MachO.open("#{bin}/fontforge")
        macho.change_dylib("@rpath/libfontforge.4.dylib", "#{lib}/libfontforge.4.dylib")
        macho.write!
```

In both cases, FontForge crashes at launch:

```
[1]    60618 killed     fontforge
```

although with a different error message than before before the fix:

```
dyld: Library not loaded: @rpath/libfontforge.4.dylib
  Referenced from: /opt/homebrew/bin/fontforge
  Reason: image not found
[1]    51706 abort      fontforge
```

it's probably something trivial, but I'm not familiar with this codebase or how dyld works

**Comment by carlocab** [comment]
I don't think using `install_name_tool` or the MachO gem is the right fix for this anyway. See the discussion at #68457.

Try setting `CMAKE_INSTALL_RPATH` or `CMAKE_INSTALL_NAME_DIR` instead.

**Comment by radex** [comment]
> Try setting CMAKE_INSTALL_RPATH or CMAKE_INSTALL_NAME_DIR instead.

I tried both, and unfortunately, it doesn't seem to do anything :(

**Comment by carlocab** [comment]
That sounds like a bug in their build scripts. There are about 40 formulae in homebrew/core that successfully make use of these flags, so it stands to reason that fontforge should support at least one of them.

Do you mind reporting this upstream? I imagine it'd be useful for them to support those CMake variables anyway for their users who want to install their software in non-standard locations.

**Comment by radex** [comment]
> Do you mind reporting this upstream? I imagine it'd be useful for them to support those CMake variables anyway for their users who want to install their software in non-standard locations.

Just bumped an existing discussion: https://github.com/fontforge/fontforge/issues/4658, hoping for a helpful response, as the initial response from upstream maintainers has been [`it's a homebrew problem`](https://github.com/fontforge/fontforge/issues/4658#issuecomment-793092478)

**Comment by carlocab** [comment]
Great, thanks for following up on that, and for opening this PR.

**Comment by carlocab** [comment]
Will be closed by #73136. Thanks for looking into this, @radex.

**Pull request closed without merging**